### PR TITLE
Cull stack depth in dumpAllThreads later on

### PR DESCRIPTION
**Pull request by mikezhang1234567890**
We need the full stacktrace from getThreadInfo function since later in the
createThreadInfoArray, we copy over MonitorInfo objects which may refer to
stack frames beyond the max depth argument passed into dumpAllThreads.

Signed-off-by: Mike Zhang <mike.h.zhang@ibm.com>

Fixes https://github.com/eclipse/openj9/issues/10796

### CODE REVIEW

**Comment by pshipton** [comment]
Pls add the test case into the test suite.

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
Won't bother launching any testing until the testcase is added.

**Comment by mikezhang1234567890** [comment]
Added a test, it is `level.extended` to go along with the rest of the ThreadMXBean tests. I essentially copied the test in the issue since that seems to reliably create a MonitorInfo object with a lock stack depth of size 2, though I'm not sure if it's always the case that there will be a monitor with a lock stack depth greater than 1.

File: test/functional/Java10andUp/src/org/openj9/test/java/lang/management/ThreadMXBean/APITestJava10.java [hunk 1]
```diff
@@ -207,6 +207,22 @@ public void testDumpLimitsWithLambdas() {
...
+	/* ThreadInfo objects contain MonitorInfo objects which carry stack depth info that can be higher than
+	 * max depth passed to dumpAllThreads. dumpAllThreads should still produce the ThreadInfo objects.
+	 * See https://github.com/eclipse/openj9/issues/10796
+	 */
+	@Test(groups = { "level.extended" })
+	public void testDumpWithMonitorDepthHigherThanLimit() {
+		try {
+			ThreadInfo[] threads = myBean.dumpAllThreads(true, true, 1);
+			for (ThreadInfo thInfo : threads) {
+					Assert.assertTrue(thInfo.getStackTrace().length <= 1, "Stack depth should be truncated to 1");
```
Comment (gacholio) [inline test/functional/Java10andUp/src/org/openj9/test/java/lang/management/ThreadMXBean/APITestJava10.java:219]: This line is over-indented.

**Comment by gacholio** [comment]
jenkins test sanity,extended zlinux jdk15

**Pull request merged**